File: ShopSphere-Frontend/src/services/orderService.js
import axiosInstance from "../api/axiosInstance";

export const createOrder = async () => {
  const response = await axiosInstance.post("/orders");

  return response.data;
};

export const getMyOrders = async () => {
  const response = await axiosInstance.get("/orders");

  return response.data;
};

export const getMyOrder = async (orderId) => {
  const response = await axiosInstance.get(`/orders/${orderId}`);

  return response.data;
};

File: ShopSphere-Backend/src/main/java/com/shopsphere/service/OrderService.java
package com.shopsphere.service;

import com.shopsphere.dto.OrderResponse;
import com.shopsphere.entity.OrderStatus;

import java.util.List;

public interface OrderService {

    OrderResponse createOrder(Long userId);

    List<OrderResponse> getMyOrders(Long userId);

    OrderResponse getMyOrder(
            Long userId,
            Long orderId);

    List<OrderResponse> getAllOrders();

    OrderResponse updateOrderStatus(
            Long orderId,
            OrderStatus status);
}